src/store/flow/utils/importHelpers.ts

```diff
@@ -27,21 +28,69 @@ export function updateResourcesForLanguageMatch(
 export function mergeFlowContainer(
   existingFlowContainer: IContext, newFlowContainer: IContext,
 ): IContext {
-  const newExistingFlowContainer = cloneDeep(existingFlowContainer)
-  // There should only be a single flow imported at a time currently.
-  // We block any more than this in the importer
-  const newFlow = get(newFlowContainer, 'flows[0]')
-  const newFlowUUID = get(newFlowContainer, 'flows[0].uuid')
-  const existingFlowIndex = findIndex(
-    newExistingFlowContainer.flows,
-    (flow) => flow.uuid === newFlowUUID,
-  )
-  if (existingFlowIndex < 0) {
-    newExistingFlowContainer.flows.push(newFlow)
-  } else {
-    newExistingFlowContainer.flows[existingFlowIndex] = newFlow
+  const clonedExistingFlowContainer = cloneDeep(existingFlowContainer)
+
+  const newFlows = newFlowContainer.flows
+
+  newFlows.forEach((newFlow) => {
+    const existingFlowIndex = findIndex(
+      clonedExistingFlowContainer.flows,
+      (flow) => flow.uuid === newFlow.uuid,
+    )
+
+    if (existingFlowIndex < 0) {
+      clonedExistingFlowContainer.flows.push(newFlow)
+    } else {
+      clonedExistingFlowContainer.flows[existingFlowIndex] = newFlow
+    }
+  })
+
+  return clonedExistingFlowContainer
+}
+
+export function createContainerFlowStack(jsonData: IContext): IFlow['uuid'][] {
+  const flows: IFlow[] = jsonData.flows
+  const visitedFlows: IFlow['uuid'][] = []
+  const flowStack: IFlow['uuid'][] = []
+
+  /*
+  * The recursive function processFlowForCreatingFlowStack takes a flow object as an argument and starts iterating over its blocks. If a block has a type of "Core.RunFlow,"
+  * it means that this block is calling another flow,
+  * so the function recursively calls itself with the child flow as an argument.
+  * This ensures that all the nested flows are processed before moving back to the parent flow.
+  *
+  * Once all the nested flows are processed, the parent flow is added to the flowStack array.
+  * */
```

Comment:
Thank you, Lakh. This JSDoc + unit tests help understand the algorithm 👍

(I guess we could say that we are looking for _the last flow in the list that doesn't have any parent_, right?)

Reply:
Yes, that's correct @bulatgab